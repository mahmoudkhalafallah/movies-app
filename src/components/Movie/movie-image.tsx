import React, { useState } from "react";
import styled from "styled-components/native";
import { IMAGE_URI_PREFIX } from "../../utils/Constants";

const imagePlaceholder = require('../../../assets/images/film-poster-placeholder.png')

const StyledImage = styled.Image`
width: ${(props) => props.horizontal? '100' : '150'};
height: ${(props) => props.horizontal? '100' : '200'};
margin-right: ${(props) => props.horizontal? '10' : '20'};
border-top-left-radius: 10;
border-top-right-radius: 10;
border-bottom-left-radius: 10;
border-bottom-right-radius: 10;
`

export const MovieImage: React.FC<{ horizontal: string, src?: string }> = ({ horizontal, src }) => {

    const [err, setErr] = useState(false)
    const source = (!err && src) ? { uri: `${IMAGE_URI_PREFIX}${src}` } : imagePlaceholder

    return <StyledImage
        source={source}
        horizontal={horizontal}
        onError={() => setErr(true)}
    />
}