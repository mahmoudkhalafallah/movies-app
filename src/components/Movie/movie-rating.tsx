import React from "react";
import SwipeableRating from 'react-native-swipeable-rating';
import { Variables } from "../../utils/Variables";
import styled from "styled-components/native";
import { StyledText } from "../Text";


const RatingContainer = styled.View`
flex-direction: row;
align-items: center;
`

const RatingValue = styled(StyledText)`
font-size: 10;
color: #bbb;
`

export const MovieRating: React.FC<{ rating: number, horizontal: boolean }> = ({ rating, horizontal }) => {

    const ratingPerFiveStars = (rating / 10) * 5
    return <RatingContainer>
        <SwipeableRating
        color={Variables.ratingStarColor}
        emptyColor={Variables.ratingStarColor}
        rating={ratingPerFiveStars}
        size={15}
        gap={0.1}
        xOffset={20}
        allowHalves={true}
        swipeable={false}
    />
        {(!horizontal) && <RatingValue>{rating}/10</RatingValue>}
    </RatingContainer>
}
