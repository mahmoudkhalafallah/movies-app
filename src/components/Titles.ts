import styled from 'styled-components/native'
import { Variables } from '../utils/Variables'
import { StyledText } from './Text'


export const SectionTitle = styled(StyledText)`
font-size: ${Variables.sectionTitleFontSize};
color: ${Variables.sectionTitleColor};
padding-bottom: 10;

`
export const MovieTitle = styled(StyledText)`
font-size: ${Variables.movieTitleFontSize};
color: ${Variables.movieTitleColor};
margin-bottom: 3;
`

