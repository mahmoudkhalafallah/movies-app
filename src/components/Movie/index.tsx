import React from "react";

import { MovieTitle } from "../Titles";
import { MovieImage } from "./movie-image";
import { MovieRating } from "./movie-rating";
import styled from "styled-components/native";
import { StyledText } from "../Text";
import { IMAGE_URI_PREFIX } from "../../utils/Constants";

const MovieContainer = styled.View`
flex-direction: ${({ isMyMovie }: any) => isMyMovie ? 'column' : 'row'};
justify-content: space-between;
align-items: ${({ isMyMovie }: any) => isMyMovie ? 'center' : 'flex-start'};
width: 100;
margin-top: 10;
margin-bottom: 10;
${({ isMyMovie, isFirst }: any) => (isMyMovie && isFirst) ? 'margin-left: 25' : 'margin-left: 10'}
${({ isMyMovie, isLast }: any) => (isMyMovie && isLast) ? 'margin-right: 20' : 'margin-right: 10'}
${({ isMyMovie }: any) => (!isMyMovie) && 'margin-left: 20'}
${({ isMyMovie, isLast }: any) => (!isMyMovie && isLast) && 'padding-bottom: 80'}
`

const MovieDetails = styled.View`
flex-direction: column;
`

const MovieOverview = styled(StyledText)`
font-size: 13;
font-family: 'Oswald-Light';
margin-top: 5;
`

const MovieDate = styled(StyledText)`
font-size: 10;
margin-top: 5;
`

export const Movie: React.FC<any> = ({ title, poster_path, vote_average, overview,
    release_date, isFirst, isLast, isMyMovie }: any) => {
    return <MovieContainer isFirst={isFirst} isLast={isLast} isMyMovie={isMyMovie}>
        <MovieImage src={isMyMovie ? poster_path : `${IMAGE_URI_PREFIX}${poster_path}`} isMyMovie={isMyMovie}></MovieImage>
        <MovieDetails style={{width: isMyMovie ? 100 : 150}}>
            <MovieTitle ellipsizeMode='tail' numberOfLines={1}>{title}</MovieTitle>
            {(vote_average != undefined) && <MovieRating rating={vote_average} showRatingValue={isMyMovie} />}
            {!isMyMovie && <>
                <MovieDate>{release_date}</MovieDate>
                <MovieOverview ellipsizeMode='tail' numberOfLines={6}>{overview}</MovieOverview>
            </>}
        </MovieDetails>
    </MovieContainer>
}