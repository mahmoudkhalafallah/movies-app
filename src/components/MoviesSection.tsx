import React, { useMemo, useCallback } from 'react'
import { FlatList, ActivityIndicator, View, Dimensions } from 'react-native'
import { Movie } from './Movie'
import { SectionTitle } from './Titles'
import { IMovie } from '../models/Movie'
import { StyledText } from './Text'
import styled from 'styled-components/native'

const windowWidth = Dimensions.get('window').width

const SectionHeader = styled.View`
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
	width: ${windowWidth - 40};
	align-self: center;
`

const NoMoviesText = styled(StyledText)`
	align-self: center;
	text-align: center;
	color: #ccc;
	width: ${windowWidth};
`

const Loader = () => (
	<ActivityIndicator size="large" style={{ marginBottom: 25 }} />
)

const NoMovies = () => <NoMoviesText>No Movies</NoMoviesText>

const renderMovie: React.FC<{
	item: IMovie
	index: number
	items: IMovie[]
	isMyMovie: boolean
}> = ({ item, index, items, isMyMovie }: any) => (
	<Movie
		{...item}
		isFirst={index === 0}
		isLast={index === items.length - 1}
		isMyMovie={isMyMovie}
	/>
)

export const MoviesSection: React.FC<{
	items: IMovie[]
	isMyMovie?: boolean
	loadMoreHandler?: (info: any) => void
	AddMovieModal?: () => Element
}> = ({ items, isMyMovie = false, loadMoreHandler, AddMovieModal }) => {
	const renderItem = useCallback(
		({ item, index }) => renderMovie({ item, index, items, isMyMovie }),
		[isMyMovie, items],
	)

	const LoadMore = useMemo(() => Loader(), [])

	return (
		<>
			<SectionHeader>
				<SectionTitle>{isMyMovie ? 'My Movies' : 'All Movies'}</SectionTitle>
				{AddMovieModal && AddMovieModal()}
			</SectionHeader>
			<FlatList
				style={{ height: isMyMovie ? 150 : 320 }}
				data={items}
				keyExtractor={(item: IMovie) => item.id.toString()}
				renderItem={renderItem}
				horizontal={isMyMovie}
				onEndReached={loadMoreHandler}
				onEndReachedThreshold={2}
				ListFooterComponent={items.length > 0 ? LoadMore : <></>}
				removeClippedSubviews={true}
				maxToRenderPerBatch={20}
				initialNumToRender={20}
				ListEmptyComponent={<NoMovies />}
			/>
		</>
	)
}
