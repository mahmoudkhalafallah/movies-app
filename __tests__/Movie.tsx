import 'react-native'
import React from 'react'
import { IMovie } from '../src/models/Movie'
import { Movie } from '../src/components/Movie'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const MovieSample: IMovie = {
	poster_path:
		'https://m.media-amazon.com/images/M/MV5BNGVjNWI4ZGUtNzE0MS00YTJmLWE0ZDctN2ZiYTk2YmI3NTYyXkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_UX182_CR0,0,182,268_AL_.jpg',
	id: 475557,
	title: 'Joker',
	vote_average: 8.4,
	overview:
		'During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.',
	release_date: '2019-10-02',
}

it('Movie renders when it is not of type "MyMovie"', () => {
	const rating = renderer
		.create(
			<Movie {...MovieSample} isMyMovie={false} isFirst={false} isLast={false} />,
		)
		.toJSON()
	expect(rating).toMatchSnapshot()
})

it('Movie renders correctly when it is of type "MyMovie"', () => {
	const rating = renderer
		.create(
			<Movie {...MovieSample} isMyMovie={true} isFirst={false} isLast={false} />,
		)
		.toJSON()
	expect(rating).toMatchSnapshot()
})

it('Movie renders correctly when it is of type "MyMovie" and is the first item in the list', () => {
	const rating = renderer
		.create(
			<Movie {...MovieSample} isMyMovie={true} isFirst={true} isLast={false} />,
		)
		.toJSON()
	expect(rating).toMatchSnapshot()
})

it('Movie renders correctly when it is of type "MyMovie" and is the last item in the list', () => {
	const rating = renderer
		.create(
			<Movie {...MovieSample} isMyMovie={true} isFirst={false} isLast={true} />,
		)
		.toJSON()
	expect(rating).toMatchSnapshot()
})
