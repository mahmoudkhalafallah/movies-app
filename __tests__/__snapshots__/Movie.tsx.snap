// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders correctly when it is of type "MyMovie" 1`] = `
<View
  isFirst={false}
  isLast={false}
  isMyMovie={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "width": 100,
      },
    ]
  }
>
  <Image
    isMyMovie={true}
    onError={[Function]}
    source={
      Object {
        "uri": "https://m.media-amazon.com/images/M/MV5BNGVjNWI4ZGUtNzE0MS00YTJmLWE0ZDctN2ZiYTk2YmI3NTYyXkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_UX182_CR0,0,182,268_AL_.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 10,
          "borderBottomRightRadius": 10,
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 100,
          "marginRight": 10,
          "width": 100,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {
          "width": 100,
        },
      ]
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Oswald-SemiBold",
            "fontSize": 15,
            "marginBottom": 3,
          },
        ]
      }
    >
      Joker
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Movie renders correctly when it is of type "MyMovie" and is the first item in the list 1`] = `
<View
  isFirst={true}
  isLast={false}
  isMyMovie={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 25,
        "marginRight": 10,
        "marginTop": 10,
        "width": 100,
      },
    ]
  }
>
  <Image
    isMyMovie={true}
    onError={[Function]}
    source={
      Object {
        "uri": "https://m.media-amazon.com/images/M/MV5BNGVjNWI4ZGUtNzE0MS00YTJmLWE0ZDctN2ZiYTk2YmI3NTYyXkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_UX182_CR0,0,182,268_AL_.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 10,
          "borderBottomRightRadius": 10,
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 100,
          "marginRight": 10,
          "width": 100,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {
          "width": 100,
        },
      ]
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Oswald-SemiBold",
            "fontSize": 15,
            "marginBottom": 3,
          },
        ]
      }
    >
      Joker
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Movie renders correctly when it is of type "MyMovie" and is the last item in the list 1`] = `
<View
  isFirst={false}
  isLast={true}
  isMyMovie={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 20,
        "marginTop": 10,
        "width": 100,
      },
    ]
  }
>
  <Image
    isMyMovie={true}
    onError={[Function]}
    source={
      Object {
        "uri": "https://m.media-amazon.com/images/M/MV5BNGVjNWI4ZGUtNzE0MS00YTJmLWE0ZDctN2ZiYTk2YmI3NTYyXkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_UX182_CR0,0,182,268_AL_.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 10,
          "borderBottomRightRadius": 10,
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 100,
          "marginRight": 10,
          "width": 100,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {
          "width": 100,
        },
      ]
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Oswald-SemiBold",
            "fontSize": 15,
            "marginBottom": 3,
          },
        ]
      }
    >
      Joker
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Movie renders when it is not of type "MyMovie" 1`] = `
<View
  isFirst={false}
  isLast={false}
  isMyMovie={false}
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 20,
        "marginRight": 10,
        "marginTop": 10,
        "width": 100,
      },
    ]
  }
>
  <Image
    isMyMovie={false}
    onError={[Function]}
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w600_and_h900_bestv2/https://m.media-amazon.com/images/M/MV5BNGVjNWI4ZGUtNzE0MS00YTJmLWE0ZDctN2ZiYTk2YmI3NTYyXkEyXkFqcGdeQXVyMTkxNjUyNQ@@._V1_UX182_CR0,0,182,268_AL_.jpg",
      }
    }
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 10,
          "borderBottomRightRadius": 10,
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 200,
          "marginRight": 20,
          "width": 150,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {
          "width": 150,
        },
      ]
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "Oswald-SemiBold",
            "fontSize": 15,
            "marginBottom": 3,
          },
        ]
      }
    >
      Joker
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 15,
                },
                Object {
                  "color": "#e8b923",
                  "marginRight": 0.1,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#bbb",
              "fontFamily": "Oswald-SemiBold",
              "fontSize": 10,
            },
          ]
        }
      >
        8.4
        /10
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Oswald-SemiBold",
            "fontSize": 10,
            "marginTop": 5,
          },
        ]
      }
    >
      2019-10-02
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={6}
      style={
        Array [
          Object {
            "fontFamily": "Oswald-Light",
            "fontSize": 13,
            "marginTop": 5,
          },
        ]
      }
    >
      During the 1980s, a failed stand-up comedian is driven insane and turns to a life of crime and chaos in Gotham City while becoming an infamous psychopathic crime figure.
    </Text>
  </View>
</View>
`;
